lib/linked_list.rb

```diff
@@ -6,16 +6,33 @@ def initialize
   end
 
   def append(data)
-    @head = Node.new(data) if !@head
-    data
+    if !@head
+      @head = Node.new(data)
+    else 
+      current_node = @head
+      while current_node.next_node
+        current_node = current_node.next_node
+      end 
+      current_node.add_next_node(data)
+    end 
+  end
+
+  def whole_list
+    elements = []
+    current_node = @head
+    while current_node.next_node
+      elements << current_node
+      current_node = current_node.next_node
+    end 
+    elements << current_node
   end
```

Comment:
This method shovels all nodes into an array, which is useful for counting/manipulating the nodes.